addons/project/models/project.py

```diff
@@ -1604,7 +1604,16 @@ def filtered_domain(self, domain):
 
     def copy_data(self, default=None):
         defaults = super().copy_data(default=default)
-        return [{k: v for k, v in default.items() if k in self.SELF_READABLE_FIELDS} for default in defaults]
+        extra_fields = []
+        if not self.env.user.has_group('base.group_portal'):
+            custom_fields = set()
+            for default in defaults:
+                custom_fields |= set(filter(lambda k: k.startswith('x_'), default.keys()))
+            extra_fields = self.env['ir.model.fields'].sudo().search([
+                ('name', 'in', list(custom_fields)),
+                ('copied', '=', True)
+            ]).mapped('name')
```

Comment:
Why querying `ir.model.fields` when the registry has all you need at almost no cost?
BTW the domain above is incorrect: you may get custom fields from other models...
```suggestion
            extra_fields = {fname
                            for fname, field in self._fields.items()
                            if field.manual and field.copy}
```

Reply:
It's not the first time I forget about `self._fields` cache :grimacing: